### PR TITLE
resolves #224

### LINKED ISSUE #224: Ingress API version v1
**What happened**:

Ingress v1beta1 is deprecated and not available in 1.22
Also, I need to set `ingressClassName` to use my ingress controller
that only listens to ingress objects with the `spec.ingressClassName` set properly.

**What you expected to happen**:

Ingress apiVersion is set correctly to `networking.k8s.io/v1` and an extra field in `values.yaml` makes the `spec.ingressClassName` configurable.

### PR DESCRIPTION
**Pull request by desy-hh**
updates ingress api version and its spec in the schduler webUI ingress template for current k8s versions.

### CODE REVIEW

**Comment by consideRatio** [comment]
I believe there may be two separate ingress resources defined in this Helm chart, can you make the update for both?

File: dask/templates/dask-scheduler-ingress.yaml [hunk 1]
```diff
@@ -24,7 +30,17 @@ spec:
     {{- end }}
     http:
       paths:
-      - backend:
-          serviceName: {{ template "dask.fullname" . }}-scheduler
-          servicePort: {{ .Values.webUI.servicePort }}
+        - path: /
+          {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
+          pathType: {{ .Values.webUI.ingress.pathType | default "ImplementationSpecific" }}
```
Comment (consideRatio) [inline dask/templates/dask-scheduler-ingress.yaml:35]: Let's not use a default value hardcoded in this template, but instead let it be part of values which needs a new entry for this.

I also think maybe Prefix is what makes most sense here?

**Comment by desy-hh** [comment]
> Hey @desy-hh maybe you could also add testing on a more modern version of k8s as well by updating the github workflow - that would validate that what you have done works I think.

Ok I added the 1.22 channel @consideRatio

**Review by jacobtomlinson — CHANGES_REQUESTED** [review]
Looks like this PR is ready but just needs the linters run.

Could you [run the linters](https://github.com/dask/helm-chart/blob/main/CONTRIBUTING.md#pre-commit)?

```
$ pip install pre-commit
$ pre-commit install
$ pre-commit run --all-files
```

**Comment by consideRatio** [comment]
I'll fix pre-commit failures in another PR!

Thank you @desy-hh!!

**Pull request merged**